### ISSUE TITLE
Branch

### ISSUE THREAD

**Issue by chickenlj** [apache/dubbo-go#2262]
At present, we have the following branches:
* [3.1](https://github.com/apache/dubbo-go)  Updated 3 weeks ago by [chickenlj](https://github.com/chickenlj)
* [3.0](https://github.com/apache/dubbo-go/tree/3.0) Updated 2 days ago by [dependabot[bot]](https://github.com/apps/dependabot)
* [master](https://github.com/apache/dubbo-go/tree/master) Updated last month by [dependabot[bot]](https://github.com/apps/dependabot)
* [dapr-support](https://github.com/apache/dubbo-go/tree/dapr-support) Updated 2 months ago by [berndverst](https://github.com/berndverst)
* [3.0-adaptive-stream](https://github.com/apache/dubbo-go/tree/3.0-adaptive-stream) Updated 3 months ago by Zhang Yepeng
* [1.5](https://github.com/apache/dubbo-go/tree/1.5) Updated 4 months ago by [caochengxiang](https://github.com/caochengxiang)
* [rest-protocol](https://github.com/apache/dubbo-go/tree/rest-protocol) Updated 8 months ago by [PhilYue](https://github.com/PhilYue)
* [feat/remoting_rocketmq](https://github.com/apache/dubbo-go/tree/feat/remoting_rocketmq) Updated 8 months ago by haohongfan
* [1.4](https://github.com/apache/dubbo-go/tree/1.4) Updated 3 years ago by [AlexStocks](https://github.com/AlexStocks)
* [1.3](https://github.com/apache/dubbo-go/tree/1.3) Updated 3 years ago by [AlexStocks](https://github.com/AlexStocks)
* [1.2](https://github.com/apache/dubbo-go/tree/1.2) Updated 4 years ago by [hxmhlt](https://github.com/hxmhlt)
* [1.1](https://github.com/apache/dubbo-go/tree/1.1) Updated 4 years ago by [AlexStocks](https://github.com/AlexStocks)

Below is a conclusion of the current status all the branches. 

The main branches under development are:
*`master`, the default branch, works as the stable or bugfix branch, responsible for the 3.0.x version.
* `3.1`, works as the major development or feature branch, as its name shows the first 3.1.0 version will come from here.
* `3.0` works for the same purpose as `master`, not necessary.

Feature branches need to be further investigated to decide the current status they are in now, and after that we can decide the following action (merge, delete or keep):
* [dapr-support](https://github.com/apache/dubbo-go/tree/dapr-support) Updated 2 months ago by [berndverst](https://github.com/berndverst)
* [3.0-adaptive-stream](https://github.com/apache/dubbo-go/tree/3.0-adaptive-stream) Updated 3 months ago by Zhang Yepeng
* [1.5](https://github.com/apache/dubbo-go/tree/1.5) Updated 4 months ago by [caochengxiang](https://github.com/caochengxiang)
* [rest-protocol](https://github.com/apache/dubbo-go/tree/rest-protocol) Updated 8 months ago by [PhilYue](https://github.com/PhilYue)
* [feat/remoting_rocketmq](https://github.com/apache/dubbo-go/tree/feat/remoting_rocketmq) Updated 8 months ago by haohongfan

The following branches are not in active development status for years, so they can be deleted and tracked by tags.
* [1.4](https://github.com/apache/dubbo-go/tree/1.4) Updated 3 years ago by [AlexStocks](https://github.com/AlexStocks)
* [1.3](https://github.com/apache/dubbo-go/tree/1.3) Updated 3 years ago by [AlexStocks](https://github.com/AlexStocks)
* [1.2](https://github.com/apache/dubbo-go/tree/1.2) Updated 4 years ago by [hxmhlt](https://github.com/hxmhlt)
* [1.1](https://github.com/apache/dubbo-go/tree/1.1) Updated 4 years ago by [AlexStocks](https://github.com/AlexStocks)

**Comment by justxuewei** [comment]
The [dapr-support](https://github.com/apache/dubbo-go/tree/dapr-support), [1.5](https://github.com/apache/dubbo-go/tree/1.5) should be reserved. About the other 1.x branches, they could be reserved as well, cuz they are not harmful and cause any inconvenience for development. As for the [rest-protocol](https://github.com/apache/dubbo-go/tree/rest-protocol) and the [feat/remoting_rocketmq](https://github.com/apache/dubbo-go/tree/feat/remoting_rocketmq), they could be removed.

**Comment by justxuewei** [comment]
I can't agree more. But GitHub uses "main" as the name of the default branch.

**Comment by chickenlj** [comment]
> The [dapr-support](https://github.com/apache/dubbo-go/tree/dapr-support), [1.5](https://github.com/apache/dubbo-go/tree/1.5) should be reserved

I agree to keep this branch as the only active 1.x branch since we haven't declared 1.x EOL and there still have many users stay on it.

> About the other 1.x branches, they could be reserved as well, cuz they are not harmful and do not cause any inconvenience for development.

I personally tend to delete all the other 1.x branches except for the 1.5 branche. We should have announced them as EOL long time ago and encourage users to upgrade to the latest 1.5 branches. But since we haven't done that we still have the responsibility to maintain them for serious bugs or vulnerabilities. But by observing their activities, for being inactive for at least 3 years, I don't see any reason for still keeping them. Even if there's one emergency bugfix happens, we can still quickly do that by checking out the relevant tags, not to say this situation almost never happens.

> As for the [rest-protocol](https://github.com/apache/dubbo-go/tree/rest-protocol) and the [feat/remoting_rocketmq](https://github.com/apache/dubbo-go/tree/feat/remoting_rocketmq), they could be removed.

This is noted, let's wait for others' opinions.

**Comment by chickenlj** [comment]
> change the default branch to master, i think.

Agree, that's what I am suggesting.

> The main branches currently under development are:
> 
> * master, the default branch, works as the stable or bugfix branch, responsible for the 3.0.x version.
> * 3.1, works as the major development or feature branch, as its name shows the first 3.1.0 version will come from here.
> * 3.0 works for the same purpose as master, not necessary.

**Comment by chickenlj** [comment]
1. Merge 3.1 into master and make master the default branch, delete 3.1 branch
2. Rename 3.0 branch to release-3.0
3. Delete 1.1, 1.2, 1.3, 1.4 branches
4. Delete rest-protocol, feat/remoting_rocketmq branches,
5. Rename 1.5 to release-1.5
6. Rename dapr-support to feature-dapr-support, 3.0-adaptive-stream to feature-3.0-adaptive-stream

The final branch layout would be like this:
* master (default)
* feature-dapr-support
* feature-adaptive-stream
* release-3.0
* release-1.5

**Comment by chickenlj** [comment]
Please notice that I have changed the branches to the following:
* main (the targeted default branch)
* 3.1 (currently the default one)
* feature-dapr-support
* feature-adaptive-stream
* release-3.0
* release-1.5

I have also announced the plan about stopping to maintain the 1.x  branch in the next few months. Please check tags for previously released versions 1.1,  1.2, 1.3, 1.4, etc.

Here's 
* The request for switching the default branch from 3.1 to main. https://issues.apache.org/jira/browse/INFRA-24390.
* The thread announcing 1.x version EOL https://lists.apache.org/thread/d6ndy6mvsxjcl3ngx6jvj1g82gvhn7th

**Comment by justxuewei** [comment]
I think `3.1` branch could be removed at present.

**Comment by chickenlj** [comment]
> I think `3.1` branch could be removed at present.

Yeah, I am removing it now.
